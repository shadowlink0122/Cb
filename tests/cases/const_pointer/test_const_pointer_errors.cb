// constポインタのエラーテストケース
include "stdio.cb"

struct Point {
    int x;
    int y;
};

int main() {
    println("=== Test 1: Reassigning const pointer to int ===");
    int a = 10;
    int b = 20;
    int* const ptr1 = &a;
    ptr1 = &b;  // エラー: const pointer cannot be reassigned
    println("ERROR: This should not execute");

    return 0;
}
