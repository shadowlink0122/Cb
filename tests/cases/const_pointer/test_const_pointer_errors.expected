=== Test 1: Reassigning const pointer to int ===
Error: Cannot reassign const pointer (T* const)
